async-http-client/src/main/scala/org/http4s/asynchttpclient/AsyncHttpClient.scala

```diff
@@ -64,7 +64,7 @@ object AsyncHttpClient {
           F.delay(
             httpClient
               .executeRequest(toAsyncRequest(req, dispatcher), asyncHandler(cb, dispatcher))
-          ).as(None)
+          ).as(Some(F.unit))
```

Comment:
Actually ... it looks like `executeRequest` returns a `ListenableFuture` which has an `abort()` method. So can we use that instead of `Some(F.unit)`?

Reply:
I was looking at that, but I wanted to keep the exact same semantics to not change too much around. I'll happily change it though if it improves things even more :smile: 

It actually has both `.abort(Throwable)` and `.cancel(Boolean)`.

> `public abstract void abort (Throwable t)`
> 
> Abort the current processing, and propagate the Throwable to the [AsyncHandler](https://asynchttpclient.github.io/async-http-client/apidocs/reference/com/ning/http/client/AsyncHandler.html) or Future

> `boolean cancel​(boolean mayInterruptIfRunning)`
> 
> Attempts to cancel execution of this task. This attempt will fail if the task has already completed, has already been cancelled, or could not be cancelled for some other reason. If successful, and this task has not started when cancel is called, this task should never run. If the task has already started, then the mayInterruptIfRunning parameter determines whether the thread executing this task should be interrupted in an attempt to stop the task.
>
> After this method returns, subsequent calls to [isDone()](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/Future.html#isDone()) will always return true. Subsequent calls to [isCancelled()](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/Future.html#isCancelled()) will always return true if this method returned true.

Not sure which one is the most appropriate for cats effect semantics.

Reply:
Oh, I missed that! Yes, `cancel(true)` is probably even better. The implementation looks good.

https://github.com/AsyncHttpClient/async-http-client/blob/7a370af58dc8895a27a14d0a81af2a3b91930651/client/src/main/java/org/asynchttpclient/netty/NettyResponseFuture.java#L174-L197

Reply:
done! I put it in `F.blocking`, I hope that makes sense :sweat_smile: